### PR TITLE
feat: Mercadopago Integration

### PR DESCRIPTION
**Pull request by drylian**
Just as they told me, I converted Mercadopago to work in the development version of cpgg, I tested the code version (requests) in the main version of cpgg, as I was unable to make the development version work, but I can guarantee that all the api will be working. If there is a problem with the API, I will resolve it and give another pull, unlike the old one that used the SDK, I preferred to use Mercadopago's own (tested) API, since the SDK generated some warnings when used.

### CODE REVIEW

**Comment by 1day2die** [comment]
This is awesome. Thanks!

**Comment by drylian** [comment]
If you need anything or find an error, just let me know and I'll fix that part.😀

File: app/Extensions/PaymentGateways/MercadoPago/MercadoPagoExtension.php [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+                Log::error('MercadoPago Payment: ' . $response->body());
+                throw new Exception('Payment failed');
+            }
+        } catch (Exception $ex) {
+            Log::error('MercadoPago Payment: ' . $ex->getMessage());
+            throw new Exception('Payment failed');
+        }
+    }
+
+    static function Checker(Request $request): void
```
Comment (IceToast) [inline app/Extensions/PaymentGateways/MercadoPago/MercadoPagoExtension.php:88]: Might want to follow the naming scheme already used in other payment gateways.

File: app/Extensions/PaymentGateways/MercadoPago/MercadoPagoSettings.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace App\Extensions\PaymentGateways\MercadoPago;
+
+use Spatie\LaravelSettings\Settings;
+
+class MercadoPagoSettings extends Settings
+{
+
+    public bool $enabled = false;
+    public ?string $access_token;
```
Comment (IceToast) [inline app/Extensions/PaymentGateways/MercadoPago/MercadoPagoSettings.php:11]: When you add the access_token as encrypted in the DB Migration it needs to be markes es encrypted in the Settings Class too.
https://github.com/spatie/laravel-settings?tab=readme-ov-file#encrypting-properties

**Review by IceToast — CHANGES_REQUESTED** [review]

**Comment by S0ly** [comment]
I fixed the conflicts that was in the readme (dont forget to pull), the new readme does not show what system are used for payment because there are too mutch, whitout speaking of the addons that add even more :)

**Comment by MrWeez** [comment]
Any news about the fixes?

**Comment by drylian** [comment]
> @drylian Any news about the fixes?
I think it's correct now

**Comment by S0ly** [comment]
@IceToast

File: app/Extensions/PaymentGateways/MercadoPago/MercadoPagoSettings.php (hunk 2 above)
Comment (IceToast) [reply app/Extensions/PaymentGateways/MercadoPago/MercadoPagoSettings.php:11]: This is still open*

Comment (IceToast) [reply app/Extensions/PaymentGateways/MercadoPago/MercadoPagoSettings.php:11]: @drylian

**Review by IceToast — APPROVED** [review]

**Comment by IceToast** [comment]
@drylian  Sorry, can you resolve that conflict to make the permission this PR adds readable.

For reference: https://github.com/Ctrlpanel-gg/panel/pull/910

**Comment by drylian** [comment]
> @drylian Sorry, can you resolve that conflict to make the permission this PR adds readable.
> 
> For reference: #910

this?

**Comment by MrWeez** [comment]
> > @drylian Sorry, can you resolve that conflict to make the permission this PR adds readable.
> > For reference: #910
> 
> this?

You need to add Readable names for your permissions, as in #910

**Comment by drylian** [comment]
> You need to add Readable names for your permissions, as in

image
I think that would be it, right? It's the only place that would need to change, the others still use arrays (views), and only this one had this change

**Comment by IceToast** [comment]
> @MrWeez
> 
> image I think that would be it, right? It's the only place that would need to change, the others still use arrays (views), and only this one had this change

Looks goot to me

**Comment by andremods1** [comment]
Como adicionar?

**Comment by MrWeez** [comment]
> Como adicionar?

This gateway is written for version 0.10, and it is incompatible with version 0.9.x

**Comment by MrWeez** [comment]
> Então não dá pra ter o Gateway do mercado pago

I don't quite understand you. Please use English, thanks.

**Comment by drylian** [comment]
> Então não dá pra ter o Gateway do mercado pago

O mercado pago ta disponível apenas para a versão development, eu fiz originalmente duas versões, uma para a 0.10 e a pra 0.9, porém não aceitaram ela por que não estão aceitando pull nela, caso queria a do 0.9 fale comigo no Discord:drylian

**Comment by andremods1** [comment]
Já te segui lá

**Comment by S0ly** [comment]
speak English or nobody will help.

**Comment by andremods1** [comment]
Ok

**Comment by drylian** [comment]
> speak English or nobody will help.

He wanted the Mercado Pago gateway for 0.9, but this code is for 0.10+, I made it for 0.9 too, but it's not on cpgg's Github because they weren't accepting pull requests in the "main" repo.

**Comment by 1day2die** [comment]
@drylian any updates?

**Comment by drylian** [comment]
> @drylian any updates?

He wanted the Mercado Pago gateway for 0.9, but this code is for 0.10+, I made it for 0.9 too, but it's not on cpgg's Github because they weren't accepting pull requests in the "main" repo.

**Comment by 1day2die** [comment]
i saw that. 
if you can extract the code you can place it on the market. many people want it actually. if you want money for it or place it free is up to you. Otherwise i would jank your code and put it for free in the market until i implement it in v010

**Comment by drylian** [comment]
> i saw that. if you can extract the code you can place it on the market. many people want it actually. if you want money for it or place it free is up to you. Otherwise i would jank your code and put it for free in the market until i implement it in v010

Honestly, if you want to do this, do it. I implemented the Mercado Pago code for people, and I really tried to put it in the official repository, but the people there didn't let me.
If you want to do it, at least tag me in the repository, so I can help

**Pull request merged**